### PR TITLE
Hide detail boxes with no data

### LINKED ISSUE #47: Optional Person details show up as blank boxes in the details panel
Potential fixes:
* Add a placeholder to indicate missing detail
* Remove the box for the detail if it is missing

### PR DESCRIPTION
**Pull request by hansstanley**
* Also adds a dedicated `PersonModulesCard` to display modules
* Closes #47

### CODE REVIEW

File: src/main/java/seedu/address/ui/body/address/PersonDetailPanel.java [hunk 1]
```diff
@@ -86,12 +88,26 @@ private Collection<Label> getTagLabels(Person person) {
...
     private Collection<Region> getDataCardCollection(Person person) {
-        return Stream.of(
+        List<Region> regions = Stream.of(
                 new PersonDetailCard.DetailCardData("Phone", person.getPhone().toString()),
                 new PersonDetailCard.DetailCardData("Address", person.getAddress().toString()),
-                new PersonDetailCard.DetailCardData("Email", person.getEmail().toString()))
+                new PersonDetailCard.DetailCardData("Email", person.getEmail().toString()),
+                new PersonDetailCard.DetailCardData("Gender", person.getGender().toString()),
+                new PersonDetailCard.DetailCardData("Race", person.getRace().toString()),
+                new PersonDetailCard.DetailCardData("Communication channels", person.getComms().toString()),
```
Comment (igezt) [inline src/main/java/seedu/address/ui/body/address/PersonDetailPanel.java:97]: Not sure if we want to call this "Preferred communication channel" or just communication channel. Any thoughts?

File: src/main/java/seedu/address/ui/body/address/PersonModulesCard.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @FXML
+    private Label title;
+    @FXML
+    private FlowPane modules;
+
+    /**
+     * Creates a {@code PersonModulesCard} with the given
+     * {@code title} and collection of {@code NusMod}s.
+     */
+    public PersonModulesCard(String title, Collection<NusMod> mods) {
```
Comment (igezt) [inline src/main/java/seedu/address/ui/body/address/PersonModulesCard.java:31]: Sick work mate

**Review by igezt — APPROVED** [review]
LGTM! Just some minor changes in phrasing but should be good to merge!

File: src/main/java/seedu/address/ui/body/address/PersonDetailPanel.java (hunk 1 above)
Comment (hansstanley) [reply src/main/java/seedu/address/ui/body/address/PersonDetailPanel.java:97]: I think if we have a way of marking channels as preferred, then we can add a `Preferred Communication Channels` section.

**Pull request merged**